opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -13,25 +13,41 @@
  */
 package opentracing;
 
+
 /**
- * Tracer is a simple, thin interface for Span creation.
+ * Tracer is a simple, thin interface for Span creation, and Span propagation into different transport formats.
  */
 public interface Tracer {
 
   /**
-   * Create, start, and return a new Span with the given `operationName`, all without specifying a
-   * parent Span that can be used to incorporate the newly-returned Span into an existing trace.
+   * Create, start, and return a new Span with the given `operationName`.
+   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
    *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
-   * Span feed = tracer.startTrace("GetFeed");
-   * Span http = tracer.startTrace("HandleHTTPRequest")
+   * Span feed = tracer.startTrace("GetFeed", null);
+   * Span http = tracer.startTrace("HandleHTTPRequest", feed)
    *                   .setTag("user_agent", req.UserAgent)
    *                   .setTag("lucky_number", 42);
    * }</pre>
    */
-  Span startTrace(String operationName);
+  Span startSpan(String operationName, /* @Nullable */ Span parent);
+
+  /** Takes two arguments:
+   *    a Span instance, and
+   *    a “carrier” object in which to inject that Span for cross-process propagation.
+   *
+   */
+  <T> void inject(Span span, T carrier);
```

Comment:
That's not my point. Consider the case where there are two raw binary formats someday.

```
  byte[] byteBuffer = ...;
  tracer.inject(someSpan, byteBuffer);
```

How does the Tracer impl determine which of the two binary formats to choose from if they both use `byte[]` as a carrier? It's underspecified.

One approach:

```
  byte[] byteBuffer = ...;
  tracer.inject(someSpan, SomeScope.FORMAT_A_ID, byteBuffer);
```

Another approach:

```
class CarrierA {
  public byte[] buffer() ...;
  public void setBuffer(byte[] b) ...;
}
...
  CarrierA carrier = ...;
  tracer.inject(someSpan, carrier);
```

Either of the above is feasible IMO (I prefer the former, but whatever). It's not reasonable to expect that `byte[]` always implies a specific format, though.